pkg/task/post/v1beta2/exec.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/ghodss/yaml"
+	post_v1beta2 "github.com/openebs/maya/pkg/apis/openebs.io/runtask/v1beta2"
+	"github.com/openebs/maya/pkg/template"
+)
+
+type PostExecutor struct {
+	// postTask holds the task's post information
+	PostTask *post_v1beta2.Post
+	Values   map[string]interface{}
+}
+
+// Builder enables building an instance of
+// post
+type Builder struct {
+	Executor *PostExecutor
+	errors   []error
+}
+
+// NewBuilder returns a new instance of Builder
+func NewBuilder() *Builder {
+	return &Builder{
+		Executor: &PostExecutor{
+			PostTask: &post_v1beta2.Post{},
+			Values:   make(map[string]interface{}),
+		},
+	}
+}
+
+func (b *Builder) WithPostInstances(context string, postYaml string,
```

Comment:
can we have this signature `WithTemplate(context, template string, values map[string]interface{})`